### ISSUE TITLE
Document the use of @HollowTransient on fields

### ISSUE THREAD

**Issue by DanielThomas** [Netflix/hollow#308]
There are cases in other JVM languages, such as delegates in Kotlin, where it's not possible to mark a field as transient. Support for a `@HollowIgnore` field annotation in addition to `transient` would allow these fields to be ignored by the mapper.

It'd also allow for legacy models to avoid marking fields transient that might need to be serialized in the legacy case, but aren't compatible with Hollow and an alternative is required (for instance you might have `@JsonIgnore` on a Hollow-only and `@HollowIgnore` on a JSON-only field).

For example, this Kotlin code:

```
private val parsedVersion: Version by lazy { Version.valueOf(version) }
```

Results in:
```
java.lang.IllegalArgumentException: Unexpected interface Lazy passed as field.
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper.<init>(HollowObjectTypeMapper.java:80)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectMapper.getTypeMapper(HollowObjectMapper.java:137)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper$MappedField.<init>(HollowObjectTypeMapper.java:329)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper$MappedField.<init>(HollowObjectTypeMapper.java:248)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper.<init>(HollowObjectTypeMapper.java:92)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectMapper.getTypeMapper(HollowObjectMapper.java:137)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper$MappedField.<init>(HollowObjectTypeMapper.java:329)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper$MappedField.<init>(HollowObjectTypeMapper.java:248)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectTypeMapper.<init>(HollowObjectTypeMapper.java:92)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectMapper.getTypeMapper(HollowObjectMapper.java:137)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectMapper.getTypeMapper(HollowObjectMapper.java:109)
	at com.netflix.hollow.core.write.objectmapper.HollowObjectMapper.initializeTypeState(HollowObjectMapper.java:105)
	at com.netflix.hollow.api.producer.HollowProducer.initializeDataModel(HollowProducer.java:276)
```

**Comment by dkoszewnik** [comment]
@DanielThomas I think this is duplicate with #21, for which the `@HollowTransient` annotation was added.

**Comment by DanielThomas** [comment]
Oh, certainly is - I didn't notice that existed. Perfect!

@toolbear something for your work on the documentation :)

**Comment by PaulSandoz** [comment]
Fixed with #316
